vault/resource_aws_auth_backend_client.go

```diff
@@ -95,6 +102,15 @@ func awsAuthBackendWrite(d *schema.ResourceData, meta interface{}) error {
 		data["secret_key"] = d.Get("secret_key").(string)
 	}
 
+	if stsRegion != "" {
+		data["sts_region"] = stsRegion
+	}
```

Comment:
I'm curious why this is handled in a separate if statement, instead of setting it in the data map definition like "sts_endpoint" (lines 92-97).

Reply:
I wasn't sure if this would show up in the plan for existing users (like a value has changed) if we try to submit something empty.

Reply:
Removed